From 5a922cc71d9db68c4d7f785558cfccddcdae7d76 Mon Sep 17 00:00:00 2001
From: Gao Jie <jzgao@dolby.com>
Date: Mon, 19 Sep 2016 10:39:58 +0800
Subject: [PATCH 1/9] Merge DAX changes

Conflicts:
    common/audioserver.te

CRs-fixed: 1081484
Change-Id: I278422370db53e1a52220222b26995318e19d90a
(cherry picked from commit a1ad06f3c26be3249af4a600db8b78510a984468)
---
 common/mediaserver.te | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/common/mediaserver.te b/common/mediaserver.te
index 89f5d00..e8e100a 100644
--- a/common/mediaserver.te
+++ b/common/mediaserver.te
@@ -71,10 +71,6 @@ binder_call(mediaserver, bootanim);
 
 allow mediaserver surfaceflinger:unix_stream_socket rw_socket_perms;
 
-# DOLBY_START
-set_prop(mediaserver, dolby_prop)
-# DOLBY_END
-
 # Allow mediaserver to search and get the widevine, playready firmwares
 allow mediaserver firmware_file:dir search;
 allow mediaserver firmware_file:file { read getattr open };
-- 
2.7.4


From 9f566995e02a17216d76ad9b985ddc3b25180bf0 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mikeioannina@cyanogenmod.org>
Date: Sun, 4 Dec 2016 20:45:12 +0200
Subject: [PATCH 2/9] Move bluetooth_loader sepolicy to common

* This can be also used by 8916 devices and possibly others,
  move it out of the msm8974 folder

Change-Id: Id36e890de99499dc964d88bbec9a7b6fbf758bca
---
 common/bluetooth_loader.te | 36 ++++++++++++++++++++++++++++++++++++
 common/file_contexts       |  3 +++
 msm8974/file_contexts      |  3 +++
 3 files changed, 42 insertions(+)
 create mode 100644 common/bluetooth_loader.te

diff --git a/common/bluetooth_loader.te b/common/bluetooth_loader.te
new file mode 100644
index 0000000..b935b32
--- /dev/null
+++ b/common/bluetooth_loader.te
@@ -0,0 +1,36 @@
+# Bluetooth executables and scripts
+type bluetooth_loader, domain, domain_deprecated;
+type bluetooth_loader_exec, exec_type, file_type;
+
+# Start bdAddrLoader from init
+init_daemon_domain(bluetooth_loader)
+
+# Run init.qcom.bt.sh
+allow bluetooth_loader shell_exec:file { entrypoint getattr read };
+allow bluetooth_loader bluetooth_loader_exec:file { getattr open execute_no_trans };
+
+# init.qcom.bt.sh needs /system/bin/log access
+allow bluetooth_loader devpts:chr_file rw_file_perms;
+
+# Run hci_qcomm_init from init.qcom.bt.sh
+domain_auto_trans(bluetooth_loader, hci_attach_exec, hci_attach)
+allow hci_attach bluetooth_loader:fd use;
+
+# Set persist.service.bdroid.* and bluetooth.* property values
+set_prop(bluetooth_loader, bluetooth_prop)
+
+# Allow getprop/setprop for init.qcom.bt.sh
+allow bluetooth_loader system_file:file execute_no_trans;
+allow bluetooth_loader toolbox_exec:file rx_file_perms;
+
+# Access the smd device
+allow bluetooth_loader hci_attach_dev:chr_file rw_file_perms;
+
+# And qmuxd
+allow bluetooth_loader qmuxd_socket:dir { write add_name remove_name search };
+allow bluetooth_loader qmuxd_socket:sock_file { create setattr getattr write unlink };
+allow bluetooth_loader qmuxd:unix_stream_socket { connectto };
+
+userdebug_or_eng(`
+  diag_use(bluetooth_loader)
+')
diff --git a/common/file_contexts b/common/file_contexts
index c216da9..b8fb567 100644
--- a/common/file_contexts
+++ b/common/file_contexts
@@ -30,6 +30,7 @@
 /dev/sdsprpc-smd                                u:object_r:dsp_device:s0
 /dev/sensors                                    u:object_r:sensors_device:s0
 /dev/smd.*                                      u:object_r:smd_device:s0
+/dev/smd3                                       u:object_r:hci_attach_dev:s0
 /dev/smem_log                                   u:object_r:smem_log_device:s0
 /dev/ttyHSL0                                    u:object_r:console_device:s0
 /dev/ttyMSM0                                    u:object_r:console_device:s0
@@ -253,6 +254,7 @@
 /system/bin/fstman                              u:object_r:fstman_exec:s0
 /system/vendor/bin/mdtpd                        u:object_r:mdtpdaemon_exec:s0
 /system/bin/wifi_ftmd                           u:object_r:wifi_ftmd_exec:s0
+/system/etc/init\.qcom\.bt\.sh                  u:object_r:bluetooth_loader_exec:s0
 
 ###################################
 # sysfs files
@@ -352,6 +354,7 @@
 # persist files
 #
 /persist(/.*)?                                                      u:object_r:persist_file:s0
+/persist/\.bt_nv\.bin                                               u:object_r:bluetooth_data_file:s0
 /persist/bluetooth(/.*)?                                            u:object_r:persist_bluetooth_file:s0
 /persist/drm(/.*)?                                                  u:object_r:persist_drm_file:s0
 /persist/sensors(/.*)?                                              u:object_r:sensors_persist_file:s0
diff --git a/msm8974/file_contexts b/msm8974/file_contexts
index 0177997..0cf6b6e 100644
--- a/msm8974/file_contexts
+++ b/msm8974/file_contexts
@@ -39,3 +39,6 @@
 /dev/block/platform/msm_sdcc\.1/by-name/logdump               u:object_r:logdump_partition:s0
 /dev/block/mmcblk0                                            u:object_r:root_block_device:s0
 /dev/block/mmcblk0rpmb                                        u:object_r:rpmb_device:s0
+
+# mpdecision
+/data/misc/perfd(/.*)?                                        u:object_r:mpctl_socket:s0
\ No newline at end of file
-- 
2.7.4


From 0dfaa130b8a79b3ff57b97a3574bd1fb70cb6684 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mikeioannina@cyanogenmod.org>
Date: Sun, 4 Dec 2016 21:05:56 +0200
Subject: [PATCH 3/9] Readd support for old perfd socket

* This is still used by older 8916 blobs and possibly others,
  move the label out of the msm8974 folder and readd the old rules
* This essentially reverts 3173a8c8b8a93e55403f65e60b31d937ea8a18b4

Change-Id: I5c51052b3622e63a14f74600cdbb76d515001841
---
 common/file_contexts  | 1 +
 common/perfd.te       | 3 ++-
 msm8974/file_contexts | 5 +----
 3 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/common/file_contexts b/common/file_contexts
index b8fb567..ea7796d 100644
--- a/common/file_contexts
+++ b/common/file_contexts
@@ -318,6 +318,7 @@
 /data/time(/.*)?                                                    u:object_r:time_data_file:s0
 /data/nfc(/.*)?                                                     u:object_r:nfc_data_file:s0
 /data/system/perfd(/.*)?                                            u:object_r:mpctl_data_file:s0
+/data/misc/perfd(/.*)?                                              u:object_r:mpctl_socket:s0
 /data/misc/iop(/.*)?                                                u:object_r:iop_data_file:s0
 /data/misc/display(/.*)?                                            u:object_r:display_misc_file:s0
 /data/misc/ipa(/.*)?                                                u:object_r:ipacm_data_file:s0
diff --git a/common/perfd.te b/common/perfd.te
index fb85729..bde7324 100644
--- a/common/perfd.te
+++ b/common/perfd.te
@@ -14,7 +14,8 @@ allow perfd {
 allow perfd self:{ netlink_kobject_uevent_socket socket} create_socket_perms;
 
 # mpctl socket
-allow perfd mpctl_socket:sock_file rw_file_perms;
+allow perfd mpctl_socket:dir rw_dir_perms;
+allow perfd mpctl_socket:sock_file create_file_perms;
 
 # default_values file
 allow perfd mpctl_data_file:dir rw_dir_perms;
diff --git a/msm8974/file_contexts b/msm8974/file_contexts
index 0cf6b6e..8a9bb71 100644
--- a/msm8974/file_contexts
+++ b/msm8974/file_contexts
@@ -38,7 +38,4 @@
 /dev/block/platform/msm_sdcc\.1/by-name/userdata              u:object_r:userdata_block_device:s0
 /dev/block/platform/msm_sdcc\.1/by-name/logdump               u:object_r:logdump_partition:s0
 /dev/block/mmcblk0                                            u:object_r:root_block_device:s0
-/dev/block/mmcblk0rpmb                                        u:object_r:rpmb_device:s0
-
-# mpdecision
-/data/misc/perfd(/.*)?                                        u:object_r:mpctl_socket:s0
\ No newline at end of file
+/dev/block/mmcblk0rpmb                                        u:object_r:rpmb_device:s0
\ No newline at end of file
-- 
2.7.4


From c4366846a9740855100673888e7d554c6e85262c Mon Sep 17 00:00:00 2001
From: Michael Bestas <mikeioannina@cyanogenmod.org>
Date: Sun, 4 Dec 2016 22:10:57 +0200
Subject: [PATCH 4/9] Allow qmuxd access diag

* Lots of older blobs need this, move the rule out
  of the 8974 folder

Change-Id: I6cf27622e7e93dd8addb53ecf5443f0e3222d9bb
---
 common/qmuxd.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common/qmuxd.te b/common/qmuxd.te
index 35d6f1f..9ca4b03 100644
--- a/common/qmuxd.te
+++ b/common/qmuxd.te
@@ -6,6 +6,7 @@ init_daemon_domain(qmuxd)
 userdebug_or_eng(`
   domain_auto_trans(shell, qmuxd_exec, qmuxd)
   domain_auto_trans(adbd, qmuxd_exec, qmuxd)
+  diag_use(qmuxd)
 ')
 
 #Allow qmuxd to operate on various qmux device sockets
-- 
2.7.4


From a83695beab4d5b08911cd1ee51b6ce8baf2b3a83 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Tue, 6 Dec 2016 05:28:02 -0800
Subject: [PATCH 5/9] Allow hci_qcomm_init /persist/.bt_nv.bin access

Change-Id: I332e25249d5e6e74ae1a082f6e9928e3b1fb11af
---
 common/bluetooth_loader.te | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/common/bluetooth_loader.te b/common/bluetooth_loader.te
index b935b32..e1d5be3 100644
--- a/common/bluetooth_loader.te
+++ b/common/bluetooth_loader.te
@@ -23,6 +23,10 @@ set_prop(bluetooth_loader, bluetooth_prop)
 allow bluetooth_loader system_file:file execute_no_trans;
 allow bluetooth_loader toolbox_exec:file rx_file_perms;
 
+# Allow hci_qcomm_init /persist/.bt_nv.bin access
+r_dir_file(bluetooth_loader, persist_file);
+allow bluetooth_loader bluetooth_data_file:file r_file_perms;
+
 # Access the smd device
 allow bluetooth_loader hci_attach_dev:chr_file rw_file_perms;
 
-- 
2.7.4


From 974627f287c170a0d57f2bcdcf9e448c0f749676 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mikeioannina@cyanogenmod.org>
Date: Mon, 5 Dec 2016 13:10:08 +0200
Subject: [PATCH 6/9] Add common rules for random MAC generation

* This is seen in many devices, put it here to avoid
  duplication in device trees

Change-Id: I5596eb500502adcff7ba79cd2c478eab0715a3cf
---
 common/file_contexts    | 1 +
 common/wcnss_service.te | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/common/file_contexts b/common/file_contexts
index ea7796d..7e8c87e 100644
--- a/common/file_contexts
+++ b/common/file_contexts
@@ -356,6 +356,7 @@
 #
 /persist(/.*)?                                                      u:object_r:persist_file:s0
 /persist/\.bt_nv\.bin                                               u:object_r:bluetooth_data_file:s0
+/persist/\.genmac                                                   u:object_r:wifi_data_file:s0
 /persist/bluetooth(/.*)?                                            u:object_r:persist_bluetooth_file:s0
 /persist/drm(/.*)?                                                  u:object_r:persist_drm_file:s0
 /persist/sensors(/.*)?                                              u:object_r:sensors_persist_file:s0
diff --git a/common/wcnss_service.te b/common/wcnss_service.te
index 05b31d4..64e5217 100644
--- a/common/wcnss_service.te
+++ b/common/wcnss_service.te
@@ -53,3 +53,5 @@ diag_use(wcnss_service)
 
 binder_use(wcnss_service)
 use_per_mgr(wcnss_service)
+
+type_transition wcnss_service persist_file:file wifi_data_file ".genmac";
-- 
2.7.4


From 46bb91a540ec05926d9fd6f948c49b0976aeee3e Mon Sep 17 00:00:00 2001
From: Ricardo Cerqueira <ricardo@cyngn.com>
Date: Wed, 13 Jul 2016 16:54:59 +0100
Subject: [PATCH 7/9] wcnss: Fix I40eeb6895f1c5550813bf8b4182e33f9a4dc5dfd

This applies to user builds, too

Change-Id: Iddc7421ba83b3fd7b5f9a66b2dd1c3ffe240e1fa
---
 common/wcnss_service.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/common/wcnss_service.te b/common/wcnss_service.te
index 64e5217..26f8d7e 100644
--- a/common/wcnss_service.te
+++ b/common/wcnss_service.te
@@ -32,6 +32,7 @@ allow wcnss_service self:netlink_generic_socket create_socket_perms;
 allow wcnss_service firmware_file:dir r_dir_perms;
 allow wcnss_service firmware_file:file r_file_perms;
 allow wcnss_service sysfs:file w_file_perms;
+allow wcnss_service storage_file:dir search;
 
 # allow access to netd
 unix_socket_connect(wcnss_service, netd, netd)
-- 
2.7.4


From 7ca82ee144409fbf91977c93ae97101a24faa561 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mikeioannina@cyanogenmod.org>
Date: Mon, 5 Dec 2016 13:37:01 +0200
Subject: [PATCH 8/9] Escape '.' character

Change-Id: I3d3c1c2d0c04f8dc77037cbf47ce7b1452fe8c33
---
 common/file_contexts  |  6 ++--
 msm8909/file_contexts | 40 +++++++++++------------
 msm8916/file_contexts | 28 ++++++++--------
 msm8937/file_contexts | 32 +++++++++----------
 msm8952/file_contexts | 46 +++++++++++++--------------
 msm8953/file_contexts | 32 +++++++++----------
 msm8976/file_contexts |  2 +-
 msm8992/file_contexts | 24 +++++++-------
 msm8994/file_contexts | 48 ++++++++++++++--------------
 msm8996/file_contexts | 88 +++++++++++++++++++++++++--------------------------
 msm8998/file_contexts | 33 +++++++++----------
 11 files changed, 188 insertions(+), 191 deletions(-)

diff --git a/common/file_contexts b/common/file_contexts
index 7e8c87e..077fb33 100644
--- a/common/file_contexts
+++ b/common/file_contexts
@@ -167,8 +167,8 @@
 /system/bin/netmgrd                             u:object_r:netmgrd_exec:s0
 /system/bin/qmuxd                               u:object_r:qmuxd_exec:s0
 /system/bin/port-bridge                         u:object_r:port-bridge_exec:s0
-/system/bin/sensors.qcom                        u:object_r:sensors_exec:s0
-#/system/bin/sns.*                               u:object_r:sensors_test_exec:s0
+/system/bin/sensors\.qcom                       u:object_r:sensors_exec:s0
+/system/bin/sns.*                               u:object_r:sensors_test_exec:s0
 /system/bin/test_diag                           u:object_r:diag_exec:s0
 /system/vendor/bin/thermal-engine               u:object_r:thermal-engine_exec:s0
 /system/bin/vm_bms                              u:object_r:vm_bms_exec:s0
@@ -247,7 +247,7 @@
 /system/bin/usf_sw_calib                        u:object_r:usf_exec:s0
 /system/bin/usf_pairing                         u:object_r:usf_exec:s0
 /system/bin/usf_tester                          u:object_r:usf_exec:s0
-/system/vendor/bin/RIDLClient.exe               u:object_r:RIDL_exec:s0
+/system/vendor/bin/RIDLClient\.exe              u:object_r:RIDL_exec:s0
 /system/vendor/bin/LKCore                       u:object_r:qti_logkit_exec:s0
 /system/bin/tbaseLoader                         u:object_r:tbaseLoader_exec:s0
 /system/bin/mcStarter                           u:object_r:mcStarter_exec:s0
diff --git a/msm8909/file_contexts b/msm8909/file_contexts
index bbebd72..9a4a10c 100644
--- a/msm8909/file_contexts
+++ b/msm8909/file_contexts
@@ -27,27 +27,27 @@
 ###################################
 # Primary storage device nodes
 #
-/dev/block/platform/soc.0/7824900.sdhci/by-name/boot                  u:object_r:boot_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/recovery              u:object_r:recovery_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/fsg                   u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/fsc                   u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/modemst1              u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/modemst2              u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/ssd                   u:object_r:ssd_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/misc                  u:object_r:misc_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/userdata              u:object_r:userdata_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/logdump               u:object_r:logdump_partition:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/cache                 u:object_r:cache_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/config                u:object_r:frp_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/boot                u:object_r:boot_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/recovery            u:object_r:recovery_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/fsg                 u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/fsc                 u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/modemst1            u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/modemst2            u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/ssd                 u:object_r:ssd_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/misc                u:object_r:misc_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/userdata            u:object_r:userdata_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/logdump             u:object_r:logdump_partition:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/cache               u:object_r:cache_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/config              u:object_r:frp_block_device:s0
 /dev/block/mmcblk0                                                    u:object_r:root_block_device:s0
 /dev/block/mmcblk0rpmb                                                u:object_r:rpmb_device:s0
 
 # for wear we are still using soc not soc.0
-/dev/block/platform/soc/7824900.sdhci/by-name/fsg                   u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/fsc                   u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/modemst1              u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/modemst2              u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/ssd                   u:object_r:ssd_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/misc                  u:object_r:misc_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/userdata              u:object_r:userdata_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/logdump               u:object_r:logdump_partition:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/fsg                  u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/fsc                  u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/modemst1             u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/modemst2             u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/ssd                  u:object_r:ssd_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/misc                 u:object_r:misc_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/userdata             u:object_r:userdata_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/logdump              u:object_r:logdump_partition:s0
diff --git a/msm8916/file_contexts b/msm8916/file_contexts
index b92c365..0c92e95 100644
--- a/msm8916/file_contexts
+++ b/msm8916/file_contexts
@@ -28,19 +28,19 @@
 ###################################
 # Primary storage device nodes
 #
-/dev/block/platform/soc.0/7824900.sdhci/by-name/boot                  u:object_r:boot_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/recovery              u:object_r:recovery_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/fsg                   u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/fsc                   u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/modemst1              u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/modemst2              u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/ssd                   u:object_r:ssd_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/misc                  u:object_r:misc_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/userdata              u:object_r:userdata_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/logdump               u:object_r:logdump_partition:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/config                u:object_r:frp_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/fsg                 u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/fsc                 u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/modemst1            u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/modemst2            u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/ssd                 u:object_r:ssd_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/misc                u:object_r:misc_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/userdata            u:object_r:userdata_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/logdump             u:object_r:logdump_partition:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/config              u:object_r:frp_block_device:s0
 /dev/block/mmcblk0                                                    u:object_r:root_block_device:s0
 /dev/block/mmcblk0rpmb                                                u:object_r:rpmb_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/frp                   u:object_r:frp_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/cache                 u:object_r:cache_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/system                u:object_r:system_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/frp                 u:object_r:frp_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/boot                u:object_r:boot_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/recovery            u:object_r:recovery_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/system              u:object_r:system_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/cache               u:object_r:cache_block_device:s0
diff --git a/msm8937/file_contexts b/msm8937/file_contexts
index 5aea987..16c6a6d 100644
--- a/msm8937/file_contexts
+++ b/msm8937/file_contexts
@@ -32,24 +32,24 @@
 /dev/block/mmcblk0rpmb                                              u:object_r:rpmb_device:s0
 
 #Using soc instead of soc.0 for 3.18 kernel
-/dev/block/platform/soc/7824900.sdhci/by-name/fsg                   u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/fsc                   u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/modemst1              u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/modemst2              u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/ssd                   u:object_r:ssd_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/misc                  u:object_r:misc_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/system                u:object_r:system_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/userdata              u:object_r:userdata_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/dip                   u:object_r:dip_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/mdtp                  u:object_r:mdtp_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/config                u:object_r:frp_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/boot                  u:object_r:boot_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/recovery              u:object_r:recovery_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/cache                 u:object_r:cache_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/logdump               u:object_r:logdump_partition:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/fsg                  u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/fsc                  u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/modemst1             u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/modemst2             u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/ssd                  u:object_r:ssd_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/misc                 u:object_r:misc_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/system               u:object_r:system_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/userdata             u:object_r:userdata_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/dip                  u:object_r:dip_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/mdtp                 u:object_r:mdtp_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/config               u:object_r:frp_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/boot                 u:object_r:boot_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/recovery             u:object_r:recovery_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/cache                u:object_r:cache_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/logdump              u:object_r:logdump_partition:s0
 
 
 
 #rawdump partition
-/dev/block/platform/soc/7824900.sdhci/by-name/rawdump               u:object_r:rawdump_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/rawdump              u:object_r:rawdump_block_device:s0
 /sys/kernel/dload/emmc_dload                                        u:object_r:sysfs_emmc_dload:s0
diff --git a/msm8952/file_contexts b/msm8952/file_contexts
index b03d222..f4afa1c 100644
--- a/msm8952/file_contexts
+++ b/msm8952/file_contexts
@@ -28,31 +28,31 @@
 ###################################
 # Primary storage device nodes
 #
-/dev/block/platform/soc.0/7824900.sdhci/by-name/fsg                   u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/fsc                   u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/modemst1              u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/modemst2              u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/ssd                   u:object_r:ssd_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/misc                  u:object_r:misc_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/system                u:object_r:system_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/userdata              u:object_r:userdata_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/dip                   u:object_r:dip_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/mdtp                  u:object_r:mdtp_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/fsg                 u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/fsc                 u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/modemst1            u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/modemst2            u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/ssd                 u:object_r:ssd_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/misc                u:object_r:misc_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/system              u:object_r:system_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/userdata            u:object_r:userdata_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/dip                 u:object_r:dip_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/mdtp                u:object_r:mdtp_device:s0
 /dev/block/mmcblk0                                                    u:object_r:root_block_device:s0
 /dev/block/mmcblk0rpmb                                                u:object_r:rpmb_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/config                u:object_r:frp_block_device:s0
-/dev/block/platform/soc.0/7824900.sdhci/by-name/logdump               u:object_r:logdump_partition:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/config              u:object_r:frp_block_device:s0
+/dev/block/platform/soc\.0/7824900\.sdhci/by-name/logdump             u:object_r:logdump_partition:s0
 
 #Using soc instead of soc.0 to make it compatable with 3.18 kernel
 
-/dev/block/platform/soc/7824900.sdhci/by-name/fsg                   u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/fsc                   u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/modemst1              u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/modemst2              u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/ssd                   u:object_r:ssd_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/misc                  u:object_r:misc_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/system                u:object_r:system_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/userdata              u:object_r:userdata_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/dip                   u:object_r:dip_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/mdtp                  u:object_r:mdtp_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/logdump               u:object_r:logdump_partition:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/fsg                  u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/fsc                  u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/modemst1             u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/modemst2             u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/ssd                  u:object_r:ssd_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/misc                 u:object_r:misc_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/system               u:object_r:system_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/userdata             u:object_r:userdata_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/dip                  u:object_r:dip_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/mdtp                 u:object_r:mdtp_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/logdump              u:object_r:logdump_partition:s0
diff --git a/msm8953/file_contexts b/msm8953/file_contexts
index 579f698..8612779 100644
--- a/msm8953/file_contexts
+++ b/msm8953/file_contexts
@@ -32,22 +32,22 @@
 /dev/block/mmcblk0rpmb                                              u:object_r:rpmb_device:s0
 
 #Using soc instead of soc.0 for 3.18 kernel
-/dev/block/platform/soc/7824900.sdhci/by-name/fsg                   u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/fsc                   u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/modemst1              u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/modemst2              u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/ssd                   u:object_r:ssd_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/misc                  u:object_r:misc_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/system                u:object_r:system_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/userdata              u:object_r:userdata_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/dip                   u:object_r:dip_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/mdtp                  u:object_r:mdtp_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/config                u:object_r:frp_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/logdump               u:object_r:logdump_partition:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/boot                  u:object_r:boot_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/recovery              u:object_r:recovery_block_device:s0
-/dev/block/platform/soc/7824900.sdhci/by-name/cache                 u:object_r:cache_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/fsg                  u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/fsc                  u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/modemst1             u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/modemst2             u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/ssd                  u:object_r:ssd_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/misc                 u:object_r:misc_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/system               u:object_r:system_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/userdata             u:object_r:userdata_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/dip                  u:object_r:dip_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/mdtp                 u:object_r:mdtp_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/config               u:object_r:frp_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/logdump              u:object_r:logdump_partition:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/boot                 u:object_r:boot_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/recovery             u:object_r:recovery_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/cache                u:object_r:cache_block_device:s0
 
 #rawdump partition
-/dev/block/platform/soc/7824900.sdhci/by-name/rawdump               u:object_r:rawdump_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/rawdump              u:object_r:rawdump_block_device:s0
 /sys/kernel/dload/emmc_dload                                        u:object_r:sysfs_emmc_dload:s0
diff --git a/msm8976/file_contexts b/msm8976/file_contexts
index 8a9ea58..72b10f3 100644
--- a/msm8976/file_contexts
+++ b/msm8976/file_contexts
@@ -26,5 +26,5 @@
 # IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #rawdump partition
-/dev/block/platform/soc/7824900.sdhci/by-name/rawdump               u:object_r:rawdump_block_device:s0
+/dev/block/platform/soc/7824900\.sdhci/by-name/rawdump              u:object_r:rawdump_block_device:s0
 /sys/kernel/dload/emmc_dload                                        u:object_r:sysfs_emmc_dload:s0
diff --git a/msm8992/file_contexts b/msm8992/file_contexts
index f4957dd..841a6c6 100644
--- a/msm8992/file_contexts
+++ b/msm8992/file_contexts
@@ -27,17 +27,17 @@
 ###################################
 # Dev block nodes
 #
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/fsc                                u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/fsg                                u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/modemst1                           u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/modemst2                           u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/ssd                                u:object_r:ssd_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/misc                               u:object_r:misc_block_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/userdata                           u:object_r:userdata_block_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/boot                               u:object_r:boot_block_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/recovery                           u:object_r:recovery_block_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/cache                              u:object_r:cache_block_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/config                             u:object_r:frp_block_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/logdump                            u:object_r:logdump_partition:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/fsc                              u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/fsg                              u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/modemst1                         u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/modemst2                         u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/ssd                              u:object_r:ssd_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/misc                             u:object_r:misc_block_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/userdata                         u:object_r:userdata_block_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/boot                             u:object_r:boot_block_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/recovery                         u:object_r:recovery_block_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/cache                            u:object_r:cache_block_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/frp                              u:object_r:frp_block_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/logdump                          u:object_r:logdump_partition:s0
 /dev/block/mmcblk0rpmb                                                              u:object_r:rpmb_device:s0
 /dev/block/mmcblk0                                                                  u:object_r:root_block_device:s0
diff --git a/msm8994/file_contexts b/msm8994/file_contexts
index 4c07304..2957d5d 100644
--- a/msm8994/file_contexts
+++ b/msm8994/file_contexts
@@ -32,30 +32,30 @@
 /dev/block/mmcblk0                                                                  u:object_r:root_block_device:s0
 
 # UFS devices
-/dev/block/platform/soc.0/fc594000.ufshc/by-name/fsc                                u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/fc594000.ufshc/by-name/fsg                                u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/fc594000.ufshc/by-name/modemst1                           u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/fc594000.ufshc/by-name/modemst2                           u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/fc594000.ufshc/by-name/ssd                                u:object_r:ssd_device:s0
-/dev/block/platform/soc.0/fc594000.ufshc/by-name/misc                               u:object_r:misc_block_device:s0
-/dev/block/platform/soc.0/fc594000.ufshc/by-name/userdata                           u:object_r:userdata_block_device:s0
-/dev/block/platform/soc.0/fc594000.ufshc/by-name/boot                               u:object_r:boot_block_device:s0
-/dev/block/platform/soc.0/fc594000.ufshc/by-name/recovery                           u:object_r:recovery_block_device:s0
-/dev/block/platform/soc.0/fc594000.ufshc/by-name/cache                              u:object_r:cache_block_device:s0
-/dev/block/platform/soc.0/fc594000.ufshc/by-name/frp                                u:object_r:frp_block_device:s0
-/dev/block/platform/soc.0/fc594000.ufshc/by-name/logdump                            u:object_r:logdump_partition:s0
+/dev/block/platform/soc\.0/fc594000\.ufshc/by-name/fsc                              u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/fc594000\.ufshc/by-name/fsg                              u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/fc594000\.ufshc/by-name/modemst1                         u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/fc594000\.ufshc/by-name/modemst2                         u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/fc594000\.ufshc/by-name/ssd                              u:object_r:ssd_device:s0
+/dev/block/platform/soc\.0/fc594000\.ufshc/by-name/misc                             u:object_r:misc_block_device:s0
+/dev/block/platform/soc\.0/fc594000\.ufshc/by-name/userdata                         u:object_r:userdata_block_device:s0
+/dev/block/platform/soc\.0/fc594000\.ufshc/by-name/boot                             u:object_r:boot_block_device:s0
+/dev/block/platform/soc\.0/fc594000\.ufshc/by-name/recovery                         u:object_r:recovery_block_device:s0
+/dev/block/platform/soc\.0/fc594000\.ufshc/by-name/cache                            u:object_r:cache_block_device:s0
+/dev/block/platform/soc\.0/fc594000\.ufshc/by-name/frp                              u:object_r:frp_block_device:s0
+/dev/block/platform/soc\.0/fc594000\.ufshc/by-name/logdump                          u:object_r:logdump_partition:s0
 
 
 # eMMC devices
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/fsc                                u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/fsg                                u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/modemst1                           u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/modemst2                           u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/ssd                                u:object_r:ssd_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/misc                               u:object_r:misc_block_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/userdata                           u:object_r:userdata_block_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/boot                               u:object_r:boot_block_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/recovery                           u:object_r:recovery_block_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/cache                              u:object_r:cache_block_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/config                             u:object_r:frp_block_device:s0
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/logdump                            u:object_r:logdump_partition:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/fsc                              u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/fsg                              u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/modemst1                         u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/modemst2                         u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/ssd                              u:object_r:ssd_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/misc                             u:object_r:misc_block_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/userdata                         u:object_r:userdata_block_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/boot                             u:object_r:boot_block_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/recovery                         u:object_r:recovery_block_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/cache                            u:object_r:cache_block_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/frp                              u:object_r:frp_block_device:s0
+/dev/block/platform/soc\.0/f9824900\.sdhci/by-name/logdump                          u:object_r:logdump_partition:s0
diff --git a/msm8996/file_contexts b/msm8996/file_contexts
index 6c3e83b..7a31eff 100644
--- a/msm8996/file_contexts
+++ b/msm8996/file_contexts
@@ -33,50 +33,50 @@
 /dev/block/mmcblk0                                                              u:object_r:root_block_device:s0
 
 # UFS devices
-/dev/block/platform/soc/624000.ufshc/by-name/fsc                                u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/fsg                                u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/modemst1                           u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/modemst2                           u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/ssd                                u:object_r:ssd_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/misc                               u:object_r:misc_block_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/rpm                                u:object_r:rpmb_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/system                             u:object_r:system_block_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/userdata                           u:object_r:userdata_block_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/msadp                              u:object_r:mba_debug_dev:s0
-/dev/block/platform/soc/624000.ufshc/by-name/dip                                u:object_r:dip_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/mdtp                               u:object_r:mdtp_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/boot                               u:object_r:boot_block_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/recovery                           u:object_r:recovery_block_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/cache                              u:object_r:cache_block_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/frp                                u:object_r:frp_block_device:s0
-/dev/block/platform/soc/624000.ufshc/by-name/mdm1m9kefs1                        u:object_r:efs_boot_dev:s0
-/dev/block/platform/soc/624000.ufshc/by-name/mdm1m9kefs2                        u:object_r:efs_boot_dev:s0
-/dev/block/platform/soc/624000.ufshc/by-name/mdm1m9kefs3                        u:object_r:efs_boot_dev:s0
-/dev/block/platform/soc/624000.ufshc/by-name/mdm1m9kefsc                        u:object_r:efs_boot_dev:s0
-/dev/block/platform/soc/624000.ufshc/by-name/logdump                            u:object_r:logdump_partition:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/fsc                               u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/fsg                               u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/modemst1                          u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/modemst2                          u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/ssd                               u:object_r:ssd_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/misc                              u:object_r:misc_block_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/rpm                               u:object_r:rpmb_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/system                            u:object_r:system_block_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/userdata                          u:object_r:userdata_block_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/msadp                             u:object_r:mba_debug_dev:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/dip                               u:object_r:dip_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/mdtp                              u:object_r:mdtp_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/boot                              u:object_r:boot_block_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/recovery                          u:object_r:recovery_block_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/cache                             u:object_r:cache_block_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/frp                               u:object_r:frp_block_device:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/mdm1m9kefs1                       u:object_r:efs_boot_dev:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/mdm1m9kefs2                       u:object_r:efs_boot_dev:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/mdm1m9kefs3                       u:object_r:efs_boot_dev:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/mdm1m9kefsc                       u:object_r:efs_boot_dev:s0
+/dev/block/platform/soc/624000\.ufshc/by-name/logdump                           u:object_r:logdump_partition:s0
 
 # eMMC devices
-/dev/block/platform/soc/7464900.sdhci/by-name/fsc                               u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/fsg                               u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/modemst1                          u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/modemst2                          u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/ssd                               u:object_r:ssd_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/misc                              u:object_r:misc_block_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/rpm                               u:object_r:rpmb_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/system                            u:object_r:system_block_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/userdata                          u:object_r:userdata_block_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/msadp                             u:object_r:mba_debug_dev:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/dip                               u:object_r:dip_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/mdtp                              u:object_r:mdtp_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/boot                              u:object_r:boot_block_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/recovery                          u:object_r:recovery_block_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/cache                             u:object_r:cache_block_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/frp                               u:object_r:frp_block_device:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/mdm1m9kefs1                       u:object_r:efs_boot_dev:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/mdm1m9kefs2                       u:object_r:efs_boot_dev:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/mdm1m9kefs3                       u:object_r:efs_boot_dev:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/mdm1m9kefsc                       u:object_r:efs_boot_dev:s0
-/dev/block/platform/soc/7464900.sdhci/by-name/logdump                           u:object_r:logdump_partition:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/fsc                              u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/fsg                              u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/modemst1                         u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/modemst2                         u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/ssd                              u:object_r:ssd_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/misc                             u:object_r:misc_block_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/rpm                              u:object_r:rpmb_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/system                           u:object_r:system_block_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/userdata                         u:object_r:userdata_block_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/msadp                            u:object_r:mba_debug_dev:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/dip                              u:object_r:dip_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/mdtp                             u:object_r:mdtp_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/boot                             u:object_r:boot_block_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/recovery                         u:object_r:recovery_block_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/cache                            u:object_r:cache_block_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/frp                              u:object_r:frp_block_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/mdm1m9kefs1                      u:object_r:efs_boot_dev:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/mdm1m9kefs2                      u:object_r:efs_boot_dev:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/mdm1m9kefs3                      u:object_r:efs_boot_dev:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/mdm1m9kefsc                      u:object_r:efs_boot_dev:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/logdump                          u:object_r:logdump_partition:s0
 
 ###################################
 # Dev socket nodes
@@ -96,7 +96,7 @@
 #
 /sys/devices/virtual/graphics/fb([0-2])+/lineptr_value                          u:object_r:sysfs_graphics:s0
 
-/sys/devices/soc/b00000.qcom,kgsl-3d0/kgsl/kgsl-3d0/gpu_available_frequencies   u:object_r:sysfs_kgsl:s0
+/sys/devices/soc/b00000\.qcom,kgsl-3d0/kgsl/kgsl-3d0/gpu_available_frequencies  u:object_r:sysfs_kgsl:s0
 
 ###################################
 # data files
@@ -106,5 +106,5 @@
 /data/misc/qvr(/.*)?       u:object_r:qvrd_data_file:s0
 
 #rawdump partition
-/dev/block/platform/soc/7464900.sdhci/by-name/rawdump                           u:object_r:rawdump_block_device:s0
+/dev/block/platform/soc/7464900\.sdhci/by-name/rawdump                          u:object_r:rawdump_block_device:s0
 /sys/kernel/dload/emmc_dload                                                    u:object_r:sysfs_emmc_dload:s0
diff --git a/msm8998/file_contexts b/msm8998/file_contexts
index ae7ea03..e4f6450 100644
--- a/msm8998/file_contexts
+++ b/msm8998/file_contexts
@@ -28,26 +28,23 @@
 # Dev block nodes
 
 # UFS Devices
-/dev/block/platform/soc/1da4000.ufshc/by-name/system                            u:object_r:system_block_device:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/userdata                          u:object_r:userdata_block_device:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/boot                              u:object_r:boot_block_device:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/logdump                           u:object_r:logdump_partition:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/fsc                                u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/fsg                                u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/modemst1                           u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/modemst2                           u:object_r:modem_efs_partition_device:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/ssd                                u:object_r:ssd_device:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/misc                               u:object_r:misc_block_device:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/rpm                                u:object_r:rpmb_device:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/msadp                              u:object_r:mba_debug_dev:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/recovery                           u:object_r:recovery_block_device:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/cache                              u:object_r:cache_block_device:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/frp                                u:object_r:frp_block_device:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/mdtp                               u:object_r:mdtp_device:s0
-/dev/block/platform/soc/1da4000.ufshc/by-name/dip                                u:object_r:dip_device:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/system                            u:object_r:system_block_device:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/userdata                          u:object_r:userdata_block_device:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/boot                              u:object_r:boot_block_device:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/logdump                           u:object_r:logdump_partition:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/fsc                               u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/fsg                               u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/modemst1                          u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/modemst2                          u:object_r:modem_efs_partition_device:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/ssd                               u:object_r:ssd_device:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/misc                              u:object_r:misc_block_device:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/rpm                               u:object_r:rpmb_device:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/msadp                             u:object_r:mba_debug_dev:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/recovery                          u:object_r:recovery_block_device:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/cache                             u:object_r:cache_block_device:s0
 
 #rawdump partition
-/dev/block/platform/soc/1da4000.ufshc/by-name/rawdump                            u:object_r:rawdump_block_device:s0
+/dev/block/platform/soc/1da4000\.ufshc/by-name/rawdump                           u:object_r:rawdump_block_device:s0
 /sys/kernel/dload/emmc_dload                                                     u:object_r:sysfs_emmc_dload:s0
 
 ###################################
-- 
2.7.4


From 6aac5c4edf1649bbdb47638ba5ef32fa99bf83e0 Mon Sep 17 00:00:00 2001
From: sooorajjj <sooorajjj@gmail.com>
Date: Tue, 7 Feb 2017 14:36:32 +0530
Subject: [PATCH 9/9] AOSP n-7.1.1_r1 changes !

---
 sepolicy.mk             | 4 ++++
 test/qti-testscripts.te | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/sepolicy.mk b/sepolicy.mk
index bf75ef8..50f6f5f 100644
--- a/sepolicy.mk
+++ b/sepolicy.mk
@@ -2,4 +2,8 @@
 BOARD_SEPOLICY_DIRS += \
        device/qcom/sepolicy \
        device/qcom/sepolicy/common \
+       device/qcom/sepolicy/test \
        device/qcom/sepolicy/$(TARGET_BOARD_PLATFORM)
+
+-include vendor/n-7.1.1_r1/sepolicy/sepolicy.mk
+-include vendor/n-7.1.1_r1/sepolicy/qcom/sepolicy.mk
\ No newline at end of file
diff --git a/test/qti-testscripts.te b/test/qti-testscripts.te
index 4d3eadf..241a1fb 100644
--- a/test/qti-testscripts.te
+++ b/test/qti-testscripts.te
@@ -26,7 +26,7 @@
 # IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 userdebug_or_eng(`
-  type qti-testscripts, domain, domain_deprecated, mlstrustedsubject;
+  #type qti-testscripts, domain, domain_deprecated, mlstrustedsubject;
   permissive qti-testscripts;
   domain_trans(init, shell_exec, qti-testscripts)
 
-- 
2.7.4

